src/olympia/reviews/views.py

```diff
@@ -166,7 +166,11 @@ def delete(request, addon, review_id):
     review = get_object_or_404(Review.objects, pk=review_id, addon=addon)
     if not user_can_delete_review(request, review):
         raise PermissionDenied
-    review.delete(user_responsible=request.user)
+    # Log deleting moderated reviews
+    if review.editorreview:
+        review.moderator_delete(user=request.user)
```

Comment:
It calls the `delete` method on the `Review` object, but that's different from this `delete` view. You'll need to either refactor to implement the logging directly in the `Review.delete` method, or implement `delete` method in the API so that it calls the custom `Review.moderator_delete` method.

Also, the way it's done now, it will use `moderator_delete` regardless of who deletes it, including the original poster, when it's flagged. That seems wrong ?